ppx/ppx_js/lib_internal/ppx_js_internal.ml

```diff
@@ -746,6 +745,8 @@ let transform =
     inherit Ast_traverse.map as super
 
     method! expression expr =
+      input_name := expr.pexp_loc.loc_start.pos_fname;
```

Comment:
It's unclear to me that this is correct. `expr.pexp_loc` could be equal to Location.none ..

Reply:
In which cases is `expr.pexp_loc` equal to `Location.none`?

I've checked if in the build of js_of_ocaml itself it ever comes to the case that `!input_name` isn't equal to `!Location.input_name` from compiler-libs before using it (in `inside_Js`). And it isn't equal in only 1 case: `toplevel/examples/lwt_toplevel/colorize.fake.ml` / `toplevel/examples/lwt_toplevel/colorize.ml` . The latter gets generated by the former and has a directive pointing to the former at the beginning. So in that case `!input_name` is the former (`toplevel/examples/lwt_toplevel/colorize.fake.ml`) and `!Location.input_name` is the latter (`toplevel/examples/lwt_toplevel/colorize.ml`). I thought that wasn't a big issue. 

But `expr.pexp_loc` being `Location.none` might be a bigger issue. What I thought was that that would only be the case when `Location.input_name` is `_none_` itself or `_toplevel_` or similar. Is that not true?

Reply:
A location none could be generated by another ppx running before jsoo

Reply:
Good point!

Reply:
I've reverted this change now, since I don't think there's a way to access the source filename via ppxlib in a reliable way at the moment.